### ISSUE TITLE
Empty object when assigning pirateplot() output to variable

### ISSUE THREAD

**Issue by pat-s** [ndphillips/yarrr#26]
``` R
pirate <- yarrr::pirateplot(formula = weight ~ Time,
                  data = ChickWeight,
                  theme = 2,
                  quant = c(0, 1),
                  main = "pirateplot of Chicken Weights",
                  back.col = gray(.97),
                  gl.col = gray(.5))
pirate
NULL (empty)
```

Can you insert that `pirateplot()` returns something if it is assigned to an object? 

This is needed when one want to call `png()` like this

``` R
png("path/image.png", width = 8.89, height = 6.6, 
   units = "in", res = 300)
pirate
dev.off()
```

**Comment by ndphillips** [comment]
Actually I have no idea how this should work! What should the function return? Can it be anything or does the output need to be in a specific format or class for `png()` to work?

**Comment by ndphillips** [comment]
So I tried implementing this in v.0.1.2. `pirateplot()` will now return a list containing some of the main plotting arguments, as well as a dataframe of summary statistics. Does this fulfil your needs?

**Comment by pat-s** [comment]
Tbh I do not know if this is possible using the base plotting system. For `ggplot2` it is standard that a `ggplot` object is returned which can then be used inside a `png()` call like in my first example. 

This problem is not urgent as it works if one wraps the `png()` call around the `pirateplot()` call. However it would be nice if this functionality would exist because it enables to move the saving of plots to the end of a script (this is how I usually do it) instead of saving it every time when `pirateplot()` is called. 

I will check your implementation and the general problem using the base plotting system in the next days and will come back to you!

**Comment by pat-s** [comment]
I did some research and found out that one needs to call `recordPlot` first to save the last drawn plot to a variable [(link)](http://stackoverflow.com/questions/29583849/r-saving-a-plot-in-an-object). This object can then be used in further code.

```R
p <- pirateplot(formula = weight ~ Time,
                data = ChickWeight,
                pal = "gray")

p <- recordPlot() # p contains all plotting information


png("~/Desktop/test.png")
p 
dev.off()
```

I tried to integrate `recordPlot` within `pirate plot` but was not successful. I don´t know if that is even possible.
